.github/workflows/rust-tests.yml

```diff
@@ -0,0 +1,97 @@
+name: Rust Tests
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'rust-core/**'
+      - '.github/workflows/rust-tests.yml'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'rust-core/**'
+      - '.github/workflows/rust-tests.yml'
+
+jobs:
+  lint:
+    name: Lint with Clippy on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+        components: clippy
+    
+    - name: Cache cargo registry
+      uses: actions/cache@v4
+      with:
+        path: ~/.cargo/registry
+        key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+    
+    - name: Cache cargo index
+      uses: actions/cache@v4
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.lock') }}
+    
+    - name: Cache cargo build
+      uses: actions/cache@v4
+      with:
+        path: rust-core/target
+        key: ${{ runner.os }}-cargo-clippy-target-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Does the project have to be built before Clippy can run? Or is this wasted time, because it does static checks that don't require code to be compiled?